hw/ip/padctrl/doc/_index.md

```diff
@@ -114,29 +110,37 @@ A specific implementation of a pad wrapper may choose to instantiate a technolog
 It is permissible to omit the implementation of all IO attributes except input/output inversion.
 
 The generic pad wrapper must expose the following IOs and parameters, even if they are not connected internally.
-In particular, the pad attribute vector `attr_i` must contain at least `AttrDw=6` bits, even if not all attributes are supported (it is permissible to just leave them unconnected in the pad wrapper implementation).
+In particular, the pad attribute vector `attr_i` must contain at least `AttrDw=9` bits, even if not all attributes are supported (it is permissible to just leave them unconnected in the pad wrapper implementation).
 
 Parameter      | Default (Min, Max)    | Description
 ---------------|-----------------------|-----------------------------------------------------
-Impl           | "generic"             | Implementation target (e.g., "generic", "xilinx").
+Variant        | 0                     | Pad variant to be instantiated (technology-specific)
 AttrDw         | 8 (6, -)              | Width of the pad attribute vector.
-
-Note that the pad wrapper may implement a "virtual" open drain termination, where standard bidirectional pads are employed, but instead of driving the output high for a logic 1 the pad is put into tristate mode.
-
-Signal               | Direction  | Type    | Description                         | Mandatory
----------------------|------------|---------|-------------------------------------|------------
-`inout_io`           | `inout`    | `wire`  | Bidirectional inout of the pad      | yes
-`in_o`               | `output`   | `logic` | Input data signal                   | yes
-`out_i`              | `input`    | `logic` | Output data signal                  | yes
-`oe_i`               | `input`    | `logic` | Output data enable                  | yes
-`attr_i[0]`          | `input`    | `logic` | Input/output inversion              | yes
-`attr_i[1]`          | `input`    | `logic` | Open drain enable                   | yes
-`attr_i[2]`          | `input`    | `logic` | Pull-down enable                    | no
-`attr_i[3]`          | `input`    | `logic` | Pull-up enable                      | no
-`attr_i[4]`          | `input`    | `logic` | Keeper enable                       | no
-`attr_i[5]`          | `input`    | `logic` | Drive strengh (0: strong, 1: weak)  | no
-`attr_i[AttrDw-1:6]` | `input`    | `logic` | Additional (optional) attributes    | no
-
+WarlOnly       | 0 (0, 1)              | If set to 1, no pad is instantiated and only the WARL behavior of the `attr_i` bits is output via `warl_o`
```

Comment:
can you remind me again, what situation do we expect to use this function?

Reply:
ooo okay, so you use this as a constant wiring for pad attributes.  That makes sense.
I have to say though, I feel like long term this could change as we get more pad types / variants.  Because it feels like you might end up encoding that in an hjson somewhere and just template out an attribute mask instead of relying on this.

Reply:
yeah I agree. I think this is something we'll probably do as part of the `pinmux` / `padctrl` / `padring` overhaul.